torchvision/models/googlenet.py

```diff
@@ -62,11 +62,14 @@ def googlenet(pretrained=False, progress=True, **kwargs):
 class GoogLeNet(nn.Module):
     __constants__ = ['aux_logits', 'transform_input']
 
-    def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, init_weights=True,
+    def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, init_weights=None,
                  blocks=None):
         super(GoogLeNet, self).__init__()
         if blocks is None:
             blocks = [BasicConv2d, Inception, InceptionAux]
+        if init_weights is None:
+            raise ValueError('For fast initialization, init_weights (weights initialization) is set '
+                             'to None. Pass init_weights = True or False.')
```

Comment:
Instead of raising a `ValueError` (which would be a BC-breaking change), can you instead [use a `FutureWarning` as `warnings.warn(msg, FutureWarning)`](https://docs.python.org/3/library/warnings.html) or something like that, and then set `init_weights=True`? This way users won't see their code failing right-away.

I would also change the message to something a bit more clear for the end-users. Maybe something like
> The default weight initialization of GoogleNet will change in future releases of torchvision. If you wish to keep the old behavior (which leads to long initialization times due to https://github.com/scipy/scipy/issues/11299), please set init_weight=True.

Reply:
Thanks for the review @fmassa. Actually I thought you were wishing some kind of exception.
>forcing the users to be aware of it until they explicitly set the value to either True or False

We can definitely switch to using warning as at the current context it seems to be a better choice.

Reply:
@fmassa, One question If we set `init_weights=True` after giving the warning, will it make any difference, as the object construction will still take lots of time even where default weights initialization is not required. 
As you said here, maybe we can set `init_weights=False` and ask them if 
they need weights
> please set init_weight=True

What do you suggest? Thanks.

Reply:
I think we should keep the `init_weights` default to `True` for now, because we need to give users some time to update their codes. If the behavior suddenly change without at least one release in the middle, it would not be a great experience.

Reply:
Yeah, sure!! I understand.